src/components/TimespanForm.js

```diff
@@ -181,8 +181,9 @@ class TimespanForm extends Component {
 
   handleCancelClick = () => {
     // Add the edited item back to data structure
-    console.log('unEditedFormItem', this.unEdiunEditedFormItemtedItem);
-    this.props.onSubmit(this.unEditedFormItem);
+    if (this.unEditedFormItem !== undefined) {
+      this.props.onSubmit(this.unEditedFormItem);
+    }
```

Comment:
Check to see whether the opened form is for an empty item. When the user clicks on 'Add Timespan' and cancels it without filling in the form, the program throws an error without this check.